pkg/gatherers/clusterconfig/gather_lokistack.go

```diff
@@ -0,0 +1,151 @@
+package clusterconfig
+
+// nolint: dupl
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+const lokiStackResourceLimit = 20
+
+// GatherLokiStack Collects `lokistacks.loki.grafana.com` resources.
+//
+// The gatherer will collect up to 20 resources from `openshift-logging` namespace
+// and it will report errors if it finds a `LokiStack` resource in a different namespace
+// or if there are more than 20 `LokiStacks` in the `openshift-logging` namespace.
+//
+// ### API Reference
+// None
+//
+// ### Sample data
+// - docs/insights-archive-sample/namespaces/openshift-logging/.json
+//
+// ### Location in archive
+// - `namespace/openshift-logging/loki.grafana.com/lokistacks/{name}.json`
+//
+// ### Config ID
+// `clusterconfig/lokistacks
+//
+// ### Released version
+// - 4.18.0
+//
+// ### Backported versions
+// None
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherLokiStack(ctx context.Context) ([]record.Record, []error) {
+	gatherDynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherLokiStack(ctx, gatherDynamicClient)
+}
+
+func gatherLokiStack(ctx context.Context, dynamicClient dynamic.Interface) ([]record.Record, []error) {
+	klog.V(2).Info("Start LokiStack gathering")
+	loggingResourceList, err := dynamicClient.Resource(lokiStackResource).List(ctx, metav1.ListOptions{})
+
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		klog.V(2).Infof("Unable to list %s resource due to: %s", lokiStackResource, err)
+		return nil, []error{err}
+	}
+
+	var records []record.Record
+	var errs []error
+	var otherNamespaceError = false
+	var tooManyResourcesError = false
+
+	for index := range loggingResourceList.Items {
+		item := loggingResourceList.Items[index]
+
+		namespace := item.GetNamespace()
+		if namespace != "openshift-logging" {
+			if !otherNamespaceError {
+				otherNamespaceError = true
+				errs = append(errs, fmt.Errorf("found resource in an unexpected namespace"))
```

Comment:
If it is not the `openshift-logging`, probably we would want to anonymize the cluster name. If it is only in logs, maybe it's ok, but definetely not as part of the error message, as it is included in the archive.

Are you ok with just adding a log message there? If so, probably it makes ineffective your other comment about storing the namespace in a variable, as it would be used